controllers/azurecluster_controller.go

```diff
@@ -159,7 +159,7 @@ func (r *AzureClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			return reconcile.Result{}, err
 		}
 		if !scope.IsClusterNamespaceAllowed(ctx, r.Client, identity.Spec.AllowedNamespaces, azureCluster.Namespace) {
-			conditions.MarkFalse(azureCluster, clusterv1.ReadyCondition, infrav1.NamespaceNotAllowedByIdentity, clusterv1.ConditionSeverityError, "")
+			conditions.MarkFalse(azureCluster, infrav1.NetworkInfrastructureReadyCondition, infrav1.NamespaceNotAllowedByIdentity, clusterv1.ConditionSeverityError, "")
```

Comment:
the Ready condition gets set as the summary as part of PatchObject() which is called at the end of every loop https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1667/files#diff-b74c6acb4629abe58f8286da9939563cc4404e09fe9966eb61dbde9b87274189R553

this commit reverts part of #1610 which removed NetworkInfrastructureReadyCondition so that we have a generic condition for everything that isn't async yet so that the summary doesn't set Ready as "true" until all the resources are actually ready (Summary also takes NetworkInfrastructureReadyCondition as one of the conditions so Ready is true when both NetworkInfrastructureReadyCondition and all the individual resource conditions are True)

Reply:
Got it, Thanks for the explanation.